src/internal_rep.rs

```diff
@@ -1766,6 +1780,43 @@ impl ValidatedCall {
     }
 }
 
+pub type ModuleMap<'a> = AliasMap<RefCell<ValidatedModule<'a>>>;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd)]
+pub struct ValidatedModule<'a> {
+    pub name: CascadeString,
+    types: BTreeSet<&'a TypeInfo>,
+    pub validated_modules: BTreeSet<&'a RefCell<ValidatedModule<'a>>>,
+}
+
+impl<'a> Ord for ValidatedModule<'a> {
```

Comment:
Looking at docs, I see the requirement that Ord must be consistent with PartialOrd https://doc.rust-lang.org/std/cmp/trait.Ord.html and PartialOrd must be consistent with PartialEq: https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html.  I think the derivations will just compare all elements, and since Ord now only compares names (which we want), they'll be inconsistent.

So I think we need manual implementations for PartialEq and PartialOrd as well.

Then of course, there's Eq: https://doc.rust-lang.org/std/cmp/trait.Eq.html  My reading of the docs there is that we *can* derive it.  It requires that children are Eq as well, but that should be fine, and it implements no methods, so there's no need for consistency.